tests/sentry/search/events/test_builder.py

```diff
@@ -0,0 +1,121 @@
+import datetime
+import unittest
+
+from django.utils import timezone
+from snuba_sdk.column import Column
+from snuba_sdk.conditions import Condition, Op, Or
+from snuba_sdk.orderby import Direction, OrderBy
+
+from sentry.search.events.builder import QueryBuilder
+from sentry.utils.snuba import Dataset
+
+
+class QueryBuilderTest(unittest.TestCase):
+    def setUp(self):
+        self.start = datetime.datetime(2015, 5, 18, 10, 15, 1, tzinfo=timezone.utc)
+        self.end = datetime.datetime(2015, 5, 19, 10, 15, 1, tzinfo=timezone.utc)
+        self.projects = [1, 2, 3]
+        self.params = {
+            "project_id": self.projects,
+            "start": self.start,
+            "end": self.end,
+        }
+        # These conditions should always be on a query when self.params is passed
+        self.default_conditions = [
+            Condition(Column("timestamp"), Op.GTE, self.start),
+            Condition(Column("timestamp"), Op.LT, self.end),
+            Condition(Column("project_id"), Op.IN, self.projects),
+        ]
+
+    def test_simple_query(self):
+        query = QueryBuilder(
+            Dataset.Discover,
+            self.params,
+            "user.email:foo@example.com release:1.2.1",
+            ["user.email", "release"],
+        )
+
+        assert query.where == [
+            Condition(Column("email"), Op.EQ, "foo@example.com"),
+            Condition(Column("release"), Op.EQ, "1.2.1"),
+            *self.default_conditions,
+        ]
+        assert query.select == [
+            Column("email"),
+            Column("release"),
+        ]
+        query.get_snql_query().validate()
```

Comment:
I prefer asserting that the conditions we constructed are valid first before validating the snql query. This is cause I think it's clearer what to fix if you see the select changed (in this example), rather than having to figure out why your query is invalid